allennlp/common/cached_transformers.py

```diff
@@ -75,11 +75,17 @@ def strip_prefix(s):
                     )
                 override_weights = {strip_prefix(k): override_weights[k] for k in valid_keys}
 
-            transformer = AutoModel.from_pretrained(
-                model_name,
-                state_dict=override_weights,
-                **kwargs,
+            transformer = AutoModel.from_config(
+                AutoConfig.from_pretrained(
+                    model_name,
+                    **kwargs,
+                )
             )
+            # Only load the model itself if we are using distributed training
```

Comment:
What does distributed training have to do with it?

Reply:
When DistributedDataParallel or DataParallel is used, the state dict of the DistributedDataParallel/DataParallel wrapper prepends "module." to all parameters of the actual model, since the actual model is stored within the module field. 

From HuggingFace:
```
def unwrap_model(model: torch.nn.Module) -> torch.nn.Module:
    """
    Recursively unwraps a model from potential containers (as used in distributed training).

    Args:
        model (:obj:`torch.nn.Module`): The model to unwrap.
    """
    # since there could be multiple levels of wrapping, unwrap recursively
    if hasattr(model, "module"):
        return unwrap_model(model.module)
    else:
        return model
```

Do we care to account for if a pretained model was saved without removing the DataParallel wrapper?

Reply:
I see. Sounds like this is good to have. Could you put what you just wrote into the code comment though?